ArduPlane/quadplane.cpp

```diff
@@ -3853,6 +3853,13 @@ float QuadPlane::get_land_airspeed(void)
 void QuadPlane::set_desired_spool_state(AP_Motors::DesiredSpoolState state)
 {
     if (motors->get_desired_spool_state() != state) {
+        if (state == AP_Motors::DesiredSpoolState::SHUT_DOWN) {
```

Comment:
This should probably be outside the `motors->get_desired_spool_state() != state`. 

Should we be zeroing directly into motors? Relaxing the attitude controller should achieve the same thing and also keeps it in the right state.

Reply:
By the looks of it there are also several places where we might call `set_desired_spool_state` and at some point and then later in the same loop run the attitude controllers thus un-zeroing these values again.

Reply:
> This should probably be outside the `motors->get_desired_spool_state() != state`.

why? We should only need to give this demand once when we shutdown.

> Should we be zeroing directly into motors? Relaxing the attitude controller should achieve the same thing and also keeps it in the right state.

we're putting it into SHUT_DOWN, I don't think we want stability to be running, especially if the user has selected MANUAL mode, but even when going to fixed wing mode in a fwd transition, once we've selected SHUT_DOWN we really want the motors off as soon as we can.

Reply:
> why? We should only need to give this demand once when we shutdown.

Are you sure we never run the attitude controller with the motors in shutdown state? You right we should only have to give it once, but I'm not sure we don't have a edge case somewhere.

>we're putting it into SHUT_DOWN, I don't think we want stability to be running, especially if the user has selected MANUAL mode, but even when going to fixed wing mode in a fwd transition, once we've selected SHUT_DOWN we really want the motors off as soon as we can.

A hard relax of the attitude controller is the same, you give it the same target and actual, the output should be zero which is then passed to motors. The advantage of doing it this way is we don't forget to zero the attitude controllers. For example this is what copter does:
https://github.com/ArduPilot/ardupilot/blob/73eabb1fd09d1ef28c6692d6746e4eeec4d027df/ArduCopter/mode_acro.cpp#L28-L32

At least we should reset the rate I terms when we zero the motors outputs. Keeping the attitude controller in sync with the motors outputs key to removing twitches.

Reply:
I tried it and it doesn't work. This is with the I and target reset:
![image](https://user-images.githubusercontent.com/831867/159607969-613dfd3f-49fb-4216-bdaf-844514b7b34c.png)
and here with the PR change:
![image](https://user-images.githubusercontent.com/831867/159608377-de3c0c45-e1be-428d-b78f-5b8dd808ca94.png)
we could relax anyway, but it should auto-relax due to the 100ms check in QuadPlane::motors_output()